other/lcp/README.md

```diff
@@ -35,111 +35,118 @@ As the licence document is mandatory, the app must raise an error if this file i
 
-c/ Display the hint and ask the passphrase to the user. The app the calls the r2-lcp-client library (C++) with the license and the passphrase hash as parameters. The lib returns the passphrase hash if it can decrypt the license key_check, or an error if the passphrase is incorrect. Loop until the user enters the correct passphrase or quits. 
+2/ Validate the structure of the status document. If the structure is not valid, the app must jump to the step 5, as if no status link was present in the license. 
 
-If the user has entered a good passphrase for the license, store the (license id, origin, user id, passphrase hash) tuple. It will allow later for checks a/ and b/. Note that a record will only be created when the user enters a new passphrase, meaning that this storage cannot be used as a catalog of all licenses acquired by the user.  
 
-Note also that the hash algorithm may depend on the LCP profile used in the license; therefore an evolution of the lib could calculate the hash value from the clear passphrase.
+### 3/ Get an updated license if needed
 
-### 3/ Validate the license integrity
+If the license timestamp in the 'updated' object of the Status Document is more recent than the timestamp contained in the local copy of the License Document, the client MUST download the License Document again. 
 
-Before checking the license status, it's good to verify the license integrity. This is because the status triggers reading system behaviors and therefore the link to the status document must be trusted. 
+It must then validate *again* the license structure (step 1), before going to step 4, to avoid any infinite loop if the server gives buggy date information (this should not happen, thanks to EDRLab checkings).
 
-The app calls the r2-lcp-client library (createContext()) with the license, the passphrase hash and CRL as parameters.
+If the updated license is ok, the app replaces the previous copy with the new one.
 
-The r2-lcp-client library will verify the license integrity and create a DRM context, i.e.:
+Note: it implies that if the user has changed his passphrase on the provider's end, the "old" passphrase is still ok until the 'updated' timestamp of the status document is modified (e.g. after a loan return). 
 
-* check the signature of the provider certificate using the embedded root certificate;
-* check that the provider certificate is not in the CRL; 
-* check that the provider certificate was not expired when the license was last updated;
-* validate the signature of the license;
-* check the user key;
-* check the date rights. An LCP license handles a datetime start and datetime end, which must be compared with the system datetime.
-* return a "context" structure, to be used later in decryption calls.
 
-See the Readium LCP spec section 5.5 for additional details. 
+### 4/ Check the license status
```

Comment:
After our discussion I would move this after the check of the license integrity, and use the status as base for a good user message.